### PR TITLE
Remove eslint-plugin-zero-config package

### PR DESCRIPTION
**Pull request by matcouto**
The `eslint-plugin-zero-config` package hasn't been maintained for the past 3 years and it primarily serves to set some predefined configs. In this PR, I've copied all the relevant settings from `eslint-plugin-zero-config` into our `.eslintrc.js` file and removed the package, as it is no longer needed

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -2,10 +2,337 @@
...
+    "no-template-curly-in-string": "error",
+    "no-extra-semi": "error",
+    "no-inner-declarations": "error",
+    "no-unreachable-loop": "error",
+    "require-atomic-updates": "error",
+
+    // https://github.com/Exelord/eslint-plugin-zero-config/blob/master/lib/configs/-private/eslint/strict-mode.js
+    strict: 'error',
+
+    // https://github.com/Exelord/eslint-plugin-zero-config/blob/master/lib/configs/-private/eslint/stylistic-issues.js
```
Comment (timjcook) [inline .eslintrc.js]: I don't think `stylistic-issues` is included in the final config. not sure what it's there for, but I believe this section can be removed.

**Review by timjcook — COMMENTED** [review]
I ran `npm exec -- eslint --print-config index.js` in the repo dir to generate the full config file with the zero config rules baked in, to get a sense of what was changing.

All looking good, I just noticed that one section doesn't get included, so we can clean it up.

File: .eslintrc.js (hunk 1 above)
Comment (matcouto) [reply .eslintrc.js]: Awesome! Just removed those rules

**Review by timjcook — APPROVED** [review]
👍

**Pull request merged**